pkgs/by-name/ma/maelstrom-clj/package.nix

```diff
@@ -0,0 +1,56 @@
+{ stdenv
+, lib
+, fetchurl
+, makeWrapper
+, git
+, coreutils
+, jdk
+, gnuplot
+, graphviz
+,
+}:
+stdenv.mkDerivation rec {
+  pname = "maelstrom";
+  version = "0.2.3";
+
+  src = fetchurl {
+    url = "https://github.com/jepsen-io/maelstrom/releases/download/v${version}/maelstrom.tar.bz2";
```

Comment:
We should directly fetch from github via fetchFromGitHub of possible, to avoid any issues or sneak ins in to the tarball.

Reply:
I've tried building this from source and struggled. 

It seems like clojure projects don't have good support in nixpkgs. 

There's a precedent in nixpkgs to download pre-build jar files for clojure projects, for example [clojure-lsp](https://github.com/NixOS/nixpkgs/blob/ba6e3dc802f4035a3346fc747c2436198ecfa9e9/pkgs/by-name/cl/clojure-lsp/package.nix#L8).